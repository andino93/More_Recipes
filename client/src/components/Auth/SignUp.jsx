import React from 'react';
import PropTypes from 'react-proptypes';
import classnames from 'classnames';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import signUpAction from '../../actions/auth/signupAction';
import history from '../../utils/history';
import registerValidate from '../../utils/registerValidate';


/**
 *
 *
 * @export
 * @class SignUpComponent
 * @extends {React.Component}
 */
export class SignUp extends React.Component {
/**
 * Creates an instance of SignUpComponent.
 * @param {any} props 
 * @memberof SignUpComponent
 */
  constructor(props) {
    super(props);
    this.state = {
      regDetails: {
        firstName: '',
        lastName: '',
        username: '',
        email: '',
        password: '',
        confirmPassword: ''
      },
      errors: {}
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  /**
  * 
  * 
  * @memberof SignUpComponent
  * 
  * @returns {void} void
 */
  componentWillMount() {
    if (window.sessionStorage.token) {
      history.push('/');
    }
  }

  /**
   * Check Validity
   *
   * @return {event} event
   * @param {event} event
   */
  checkValidity() {
    const { errors, isValid } = registerValidate(this.state.regDetails);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
  * Handle submit
  * 
  * @return {event} event 
  * @param {event} event
  */
  handleSubmit(event) {
    event.preventDefault();
    if (this.checkValidity()) {
      this.setState({ errors: {} });
      this.props.signUpAction(this.state.regDetails);
    }
  }
  /**
  * Handle change
  * 
  * @return {event} event 
  * @param {event} event
  */
  handleChange(event) {
    const regDetails = this.state.regDetails;
    regDetails[event.target.name] = event.target.value;
    this.setState({ regDetails });
  }
  /**
  * Renders signUp component
  * 
  * @returns {XML} XML/JSX
  */
  render() {
    const { regDetails, errors } = this.state;
    const { signUpState } = this.props;

    if (signUpState.success) {
      $('.modal-backdrop').hide();
      return (
        <Redirect to={'/'}/>
      );
    }
    return (
      <div
        className="modal fade"
        id="signUpModal"
        tabIndex="-1"
        role="dialog"
        aria-labelledby="signUpModalLabel"
        aria-hidden="true">
        <div className="modal-dialog" role="document">
          <div className="modal-content simplebox blade">
            <div className="modal-header">
              <h5 className="modal-title" id="signUpModalLabel">Sign Up</h5>
              <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <form className="mt-15" onSubmit={this.handleSubmit}>
              { signUpState.fails ?
                <div id="error" className="alert alert-danger alert-dismissible">
                  <button type="button" className="close" data-dismiss="alert" aria-label="Close">
                    <span aria-hidden="true">x</span>
                  </button>
                  { signUpState.fails }
                </div> : null
              }
              <div className="modal-body">
                <div className="form-group">
                  <label htmlFor="firstName">First Name</label>
                  <input
                    type="text"
                    className={classnames('form-control',
                      { 'is-invalid': errors.firstName ? !!errors.firstName : false })
                    }
                    name="firstName"
                    aria-describedby="firstName"
                    placeholder="First name"
                    onChange={this.handleChange}
                    value={regDetails.firstName}
                  />
                  { errors.firstName ?
                    <span className="invalid-feedback">{ errors.firstName }</span> : null
                  }
                </div>

                <div className="form-group">
                  <label htmlFor="lastName">Last Name</label>
                  <input
                    type="text"
                    className={classnames('form-control',
                      { 'is-invalid': errors.lastName ? !!errors.lastName : false })
                    }
                    name="lastName"
                    aria-describedby="lastName"
                    placeholder="Last name"
                    onChange={this.handleChange}
                    value={regDetails.lastName}
                  />
                  { errors.lastName ?
                    <span className="invalid-feedback">{ errors.lastName }</span> : null
                  }
                </div>
                <div className="form-group">
                  <label htmlFor="userName">Username</label>
                  <input
                    type="text"
                    className={classnames('form-control',
                      { 'is-invalid': errors.username ? !!errors.username : false })
                    }
                    name="username"
                    aria-describedby="username"
                    placeholder="Username"
                    onChange={this.handleChange}
                    value={regDetails.username}
                  />
                  { errors.username ?
                    <span className="invalid-feedback">{ errors.username }</span> : null
                  }
                </div>

                <div className="form-group">
                  <label htmlFor="email">Email address</label>
                  <input
                    type="email"
                    className={classnames('form-control',
                      { 'is-invalid': errors.email ? !!errors.email : false })
                    }
                    name="email"
                    aria-describedby="email"
                    placeholder="Email"
                    onChange={this.handleChange}
                    value={regDetails.email}
                  />
                  { errors.email ?
                    <span className="invalid-feedback">{ errors.email }</span> : null
                  }
                </div>

                <div className="form-group">
                  <label htmlFor="password">Password</label>
                  <input
                    type="password"
                    className={classnames('form-control',
                      { 'is-invalid': errors.email ? !!errors.email : false })
                    }
                    name="password"
                    placeholder="Password"
                    onChange={this.handleChange}
                    value={regDetails.password}
                  />
                  { errors.password ?
                    <span className="invalid-feedback">{ errors.password }</span> : null
                  }
                </div>
                <div className="form-group">
                  <label htmlFor="passwordConfirm">Confirm Password</label>
                  <input
                    type="password"
                    className={classnames('form-control',
                      { 'is-invalid': errors.email ? !!errors.email : false })
                    }
                    name="confirmPassword"
                    placeholder="Confirm Password"
                    onChange={this.handleChange}
                    value={regDetails.confirmPassword}
                  />
                  { errors.confirmPassword ?
                    <span className="invalid-feedback">{ errors.confirmPassword }</span> : null
                  }
                </div>
              </div>
              <div className="modal-footer">
                <button type="submit" className="btn btn-outline-warning">Submit</button>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

SignUp.propTypes = {
  signUpState: PropTypes.object.isRequired,
  signUpAction: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  signUpState: state.signupReducer
});

const mapDispatchToProps = dispatch => 
  bindActionCreators({ signUpAction }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);
