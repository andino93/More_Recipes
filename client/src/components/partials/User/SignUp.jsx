import React from 'react';
import ReactDOM from 'react-dom';

/**
 *
 *
 * @export
 * @class SignUpComponent
 * @extends {React.Component}
 */
export default class SignUpComponent extends React.Component {
    render() {
        return (
            <div
                className="modal fade"
                id="signUpModal"
                tabIndex="-1"
                role="dialog"
                aria-labelledby="signUpModalLabel"
                aria-hidden="true">
                <div className="modal-dialog" role="document">
                    <div className="modal-content simplebox blade">
                        <div className="modal-header">
                            <h5 className="modal-title" id="signUpModalLabel">Sign Up</h5>
                            <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">
                            <form className="mt-15">
                                <div className="form-group">
                                    <label htmlFor="fullName">Full Name</label>
                                    <input
                                        type="text"
                                        className="form-control"
                                        id="fullname"
                                        aria-describedby="fullname"
                                        placeholder="Full name"/>
                                </div>
                                <div className="form-group">
                                    <label htmlFor="userName">Username</label>
                                    <input
                                        type="text"
                                        className="form-control"
                                        id="username"
                                        aria-describedby="username"
                                        placeholder="Username"/>
                                </div>
                                <div className="form-group">
                                    <label htmlFor="email">Email address</label>
                                    <input
                                        type="email"
                                        className="form-control"
                                        id="email"
                                        aria-describedby="email"
                                        placeholder="Email"/>
                                </div>
                                <div className="form-group">
                                    <label htmlFor="mobile">Mobile Number</label>
                                    <input
                                        type="text"
                                        className="form-control"
                                        id="mobile"
                                        aria-describedby="mobile"
                                        placeholder="Mobile Number"/>
                                </div>
                                <div className="form-group">
                                    <label htmlFor="password">Password</label>
                                    <input
                                        type="password"
                                        className="form-control"
                                        id="password"
                                        placeholder="Password"/>
                                </div>
                                <div className="form-group">
                                    <label htmlFor="passwordConfirm">Confirm Password</label>
                                    <input
                                        type="password"
                                        className="form-control"
                                        id="confirm"
                                        placeholder="Confirm Password"/>
                                </div>
                            </form>
                        </div>
                        <div className="modal-footer">
                            <button type="submit" className="btn btn-outline-warning">Submit</button>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}