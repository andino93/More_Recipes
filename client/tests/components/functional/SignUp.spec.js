import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import localStorageMock from '../../__mocks__/localStorageMock';
import mockData from "../../__mocks__/mockData";
import SignUp from '../../../src/components/auth/SignUp.jsx';

configure({ adapter: new Adapter() });

// noinspection JSAnnotator
window.localStorage = localStorageMock;
window.localStorage.token = mockData.token;

describe('<SignUp>', () => {
  const store = mockStore({ signupReducer: { signUpState: {} } });
  const connectedSignUp = mountWrap(SignUp, store, { history: { push() {} } });

  it('should render SignUp component', (done) => {
    expect(connectedSignUp).toMatchSnapshot();
    done();
  });
  it('should check for componentWillReceiveProps', () => {
    connectedSignUp.find('SignUp').instance().componentWillReceiveProps({
      signUpState: {
        success: '',
      },
    });
  });
  it('should simulate handleChange', (done) => {
    connectedSignUp.find('input').at(0).simulate('change');
    done();
  });
  it('should simulate handleSubmit', (done) => {
    connectedSignUp.find('form').at(0).simulate('submit', {
      preventDefault: () => {}
    });
    connectedSignUp.find('SignUp').instance().setState({
      regDetails: {
        firstName: 'Chibueze',
        lastName: 'Ayogu',
        username: 'Chebo',
        email: 'chebo@gmail.com',
        password: 'anything',
        confirmPassword: 'anything'
      }
    });
    connectedSignUp.find('form').at(0).simulate('submit', {
      preventDefault: () => {}
    });
    done();
  });
});
