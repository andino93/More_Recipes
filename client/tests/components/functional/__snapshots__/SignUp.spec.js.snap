// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp> should render SignUp component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(SignUp)
        history={
          Object {
            "push": [Function],
          }
        }
      >
        <SignUp
          history={
            Object {
              "push": [Function],
            }
          }
          signUpAction={[Function]}
          signUpState={
            Object {
              "signUpState": Object {},
            }
          }
        >
          <div
            aria-hidden="true"
            aria-labelledby="signUpModalLabel"
            className="modal fade"
            id="signUpModal"
            role="dialog"
            tabIndex="-1"
          >
            <div
              className="modal-dialog"
              role="document"
            >
              <div
                className="modal-content simplebox blade"
              >
                <div
                  className="modal-header"
                >
                  <h5
                    className="modal-title"
                    id="signUpModalLabel"
                  >
                    Sign Up
                  </h5>
                  <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      Ã—
                    </span>
                  </button>
                </div>
                <form
                  className="mt-15"
                  onSubmit={[Function]}
                >
                  <div
                    className="modal-body"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="firstName"
                      >
                        First Name
                        <span
                          className="text-danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        aria-describedby="firstName"
                        className="form-control"
                        name="firstName"
                        onChange={[Function]}
                        placeholder="First name"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="lastName"
                      >
                        Last Name 
                        <span
                          className="text-danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        aria-describedby="lastName"
                        className="form-control"
                        name="lastName"
                        onChange={[Function]}
                        placeholder="Last name"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="userName"
                      >
                        Username 
                        <span
                          className="text-danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        aria-describedby="username"
                        className="form-control"
                        name="username"
                        onChange={[Function]}
                        placeholder="Username"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="email"
                      >
                        Email address 
                        <span
                          className="text-danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        aria-describedby="email"
                        className="form-control"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
                        value=""
                      />
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="password"
                      >
                        Password 
                        <span
                          className="text-danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        className="form-control"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
                      />
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="passwordConfirm"
                      >
                        Confirm Password 
                        <span
                          className="text-danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        className="form-control"
                        name="confirmPassword"
                        onChange={[Function]}
                        placeholder="Confirm Password"
                        type="password"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="p-10 clearfix"
                  >
                    <small
                      className="form-text text-muted float-left"
                    >
                      All input fields marked
                      <span
                        className="text-danger"
                      >
                        *
                      </span>
                       are required
                    </small>
                    <button
                      className="btn btn-warning float-right"
                      id="submitSignUp"
                      type="submit"
                    >
                      Register me
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </SignUp>
      </Connect(SignUp)>
    </Router>
  </BrowserRouter>
</Provider>
`;
